src/NzbDrone.Core/Download/TrackedDownloads/TrackedDownloadService.cs

```diff
@@ -135,6 +135,7 @@ public TrackedDownload TrackDownload(DownloadClientDefinition downloadClient, Do
                     var grabbedEvent = historyItems.FirstOrDefault(v => v.EventType == EpisodeHistoryEventType.Grabbed);
 
                     trackedDownload.Indexer = grabbedEvent?.Data["indexer"];
```

Comment:
This line is suspicious to me. We use `?.` to access `Data` on the possibly-null `grabbedEvent`, but then directly index into that data with `["indexer"]` -- if `grabbedEvent` is null, won't we try to do `null["indexer"]` and fail anyway?

Note that the line I added below doesn't use `?.` -- if I did, I'd have to mark `AddedOn` as nullable.

Reply:
It seems like I don't correctly understand `?.` --
```
a = grabbedEvent?.Data;
b = a["indexer"]
```
Throws a null pointer exception, but `b = grabbedEvent?.Data["indexer"]` doesn't.
Additionally, the `.Indexer` field of the TrackedDownload isn't nullable, yet null is clearly being written in.

However, whatever's making this line work doesn't apply to `AddedOn` below -- the compiler correctly recognizes that `grabbedEvent?.AddedOn` may be null. Solution is either to mark it as nullable in TrackedDownload, or hide the write inside a `.HasValue` call. Chose to go with the first here.

Reply:
`?.` only does something when the left side of the expression is not null, it could also be written a longer like this like this: 

```
trackedDownload.AddedOn = grabbedEvent == null ? null : grabbedEvent.Date;
```

I agree that making it nullable is the correct solution since we wouldn't want to show a default date for items that Sonarr didn't grab.